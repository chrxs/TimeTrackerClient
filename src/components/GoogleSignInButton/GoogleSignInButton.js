import React from 'react'
import PropTypes from 'prop-types'

import styles from './GoogleSignInButton.scss'

const GoogleSignInButton = ({
  disabled,
  onClick,
  ...otherProps
}) => {
  return (
    <div
      className={styles.GoogleSignInButton}
      disabled={disabled}
      onClick={onClick}
      {...otherProps}
    />
  )
}

GoogleSignInButton.displayName = 'GoogleSignInButton'

GoogleSignInButton.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func
}

GoogleSignInButton.defaultProps = {
  disabled: false,
  onClick: () => {}
}

export default GoogleSignInButton
