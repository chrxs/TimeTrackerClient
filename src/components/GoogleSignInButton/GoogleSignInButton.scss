.GoogleSignInButton {
  background: url('./images/btn_google_signin_dark_normal_web@2x.png') 50% 50% no-repeat;
  background-size: contain;
  border-radius: 3px;
  cursor: pointer;
  height: 46px;
  width: 191px;

  &:hover,
  &:focus {
    background-image: url('./images/btn_google_signin_dark_focus_web@2x.png');
  }

  &:active {
    background-image: url('./images/btn_google_signin_dark_pressed_web@2x.png');
  }

  &[disabled] {
    background-image: url('./images/btn_google_signin_dark_disabled_web@2x.png');
    cursor: disabled;
  }
}
