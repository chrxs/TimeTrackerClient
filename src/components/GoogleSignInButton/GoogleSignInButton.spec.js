/* global describe, test, expect */
import React from 'react'
import GoogleSignInButton from './GoogleSignInButton'
import renderer from 'react-test-renderer'

describe('GoogleSignInButton snapshots', () => {
  test('with just label prop', () => {
    const component = renderer.create(<GoogleSignInButton />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('with all props', () => {
    const component = renderer.create(<GoogleSignInButton disabled={false} onClick={() => {}} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
