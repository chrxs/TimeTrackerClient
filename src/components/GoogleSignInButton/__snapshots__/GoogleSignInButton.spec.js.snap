// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleSignInButton snapshots with all props 1`] = `
<div
  className="GoogleSignInButton"
  disabled={false}
  onClick={[Function]}
/>
`;

exports[`GoogleSignInButton snapshots with just label prop 1`] = `
<div
  className="GoogleSignInButton"
  disabled={false}
  onClick={[Function]}
/>
`;
